### PR TITLE
feat: add NGINX rock

### PR DESCRIPTION
**Pull request by amandahla**
Applicable spec: <link>

### Overview

This PR adds a NGINX rock and pebble layer to the Maubot charm.

### Rationale

NGINX will act as a proxy.

Note:
- PRs #7 and #8 will add Ingress and PostgreSQL , providing the actual Maubot application.
- Further PR will change Maubot layer to run under daemon user, set IngressRequires properly and also add an integration test for checking the URL.

### Juju Events Changes

Add NGINX pebble ready event.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [ ] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

No CH available.

### CODE REVIEW

**Review by cbartz — DISMISSED** [review]

**Review by javierdelapuente — DISMISSED** [review]

**Review by Thanhphan1147 — APPROVED** [review]

File: nginx_rock/etc/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+  gzip_proxied any;
+  gzip_http_version 1.1;
+  gzip_types
+   application/font-woff
+   application/font-woff2
+   application/x-javascript
+   application/xml
+   application/xml+rss
+   image/png
```
Comment (weiiwang01) [inline nginx_rock/etc/nginx.conf:23]: nit: probably don't need to compress PNG files.

File: nginx_rock/etc/nginx.conf [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+  add_header X-Content-Type-Options 'nosniff';
+  add_header X-Frame-Options 'SAMEORIGIN';
+  add_header Strict-Transport-Security "max-age=31536000; includeSubdomains; preload";
+  add_header X-XSS-Protection "1; mode=block";
+
+  log_format main '$remote_addr - $remote_user [$time_local] "$request" '
+					'$status $body_bytes_sent "$http_referer" '
+					'"$http_user_agent" "$http_x_forwarded_for" "$http_x_forwarded_proto"';
+  access_log /var/log/nginx/access.log main;
```
Comment (weiiwang01) [inline nginx_rock/etc/nginx.conf:39]: Maybe the access log should be logged to stdout, as the error log is logged to stderr?

**Review by weiiwang01 — APPROVED** [review]
I think the overall is great, but I have one question related to this: why not include nginx in the maubot image and use pebble to run both the maubot service and nginx in the same container? Is there any reason for using two containers?

**Comment by amandahla** [comment]
> I think the overall is great, but I have one question related to this: why not include nginx in the maubot image and use pebble to run both the maubot service and nginx in the same container? Is there any reason for using two containers?

Thats an excellent question. I cant think of any reason for having two containers!

**Comment by amandahla** [comment]
I'll close this PR and add NGINX to Maubot. Thanks @weiiwang01 !

**Pull request closed without merging**